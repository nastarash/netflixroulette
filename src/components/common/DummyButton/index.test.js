import React from 'react';
import { DummyButton } from './index';

describe('<DummyButton />', () => {

  test('should be rendered correctly', () => {
    const btn = shallow(<DummyButton />);
    expect(btn).toMatchSnapshot();
  });

  test('has class button', () => {
    const btn = shallow(<DummyButton />);
    expect(btn.hasClass('button')).toBeTruthy();
  });

  test('active button has class active-button', () => {
    const btn = shallow(<DummyButton isActive={true} />);
    expect(btn.hasClass('button-active')).toBeTruthy();
  });

  test('inactive button has class inactive-button', () => {
    const btn = shallow(<DummyButton isActive={false} />);
    expect(btn.hasClass('button-inactive')).toBeTruthy();
  });

  test('button has width property', () => {
    const btn = mount(<DummyButton />);
    btn.setProps({ width: '500px' });
    expect(btn.props().width).toEqual('500px');
  });

});
