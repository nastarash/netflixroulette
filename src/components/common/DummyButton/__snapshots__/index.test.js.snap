// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DummyButton /> should be rendered correctly 1`] = `
<input
  className="button button-inactive"
  style={Object {}}
  type="button"
/>
`;
